tensorflow/lite/micro/micro_profiler.h

```diff
@@ -18,8 +18,25 @@ limitations under the License.
 
 #include <cstdint>
 
+#ifdef LCE_RUN_ON_FPGA
```

Comment:
this is where all the layering / interfacing / information hiding violations start. i'm aware that i mentioned i could just pass in the flags via the including project's Makefile, but that was wrong, because there is no dependency of the resulting object files on those flags changing,  meaning if i build with 2 different profiling levels in succession, i'd end up with some of the files not being re-built as they'd have to.

that's why i settled for something embarrassingly close to the original version.

the other option i tried was similar: do similar things in the enclosing project and then include that header.  that's worse, because there'd be a level of indirection that doesn't help and an additional layering violation. whereas with this PR here, at least it is somewhat contained and should build exactly the same way as before when used in a different context.